src/CairoPainter.ts

```diff
@@ -0,0 +1,103 @@
+import { CamOverlayDrawingAPI, CamOverlayDrawingOptions, CairoCreateResponse } from './CamOverlayDrawingAPI';
+import ResourceManager from './ResourceManager';
+import CairoFrame from './CairoFrame';
+
+const COORD: Record<string, [number, number]> = {
+    top_left: [-1, -1],
+    center_left: [-1, 0],
+    bottom_left: [-1, 1],
+    top_center: [0, -1],
+    center: [0, 0],
+    bottom_center: [0, 1],
+    top_right: [1, -1],
+    center_right: [1, 0],
+    bottom_right: [1, 1],
+};
+
+type Options = {
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+    screenWidth: number;
+    screenHeight: number;
+    coAlignment: string;
+};
+
+export default class CairoPainter extends CairoFrame {
+    private screenWidth: number;
+    private screenHeight: number;
+    private coAlignment: [number, number];
+
+    private surface?: string;
+    private cairo?: string;
+    private cod: CamOverlayDrawingAPI;
+    private connected = false;
+
+    constructor(opt: Options, coopt: CamOverlayDrawingOptions, rm: ResourceManager) {
+        super(opt, rm);
+        this.coAlignment = COORD[opt.coAlignment];
+        this.screenWidth = opt.screenWidth;
+        this.screenHeight = opt.screenHeight;
+
+        this.cod = new CamOverlayDrawingAPI(coopt);
+    }
+
+    async connect() {
+        this.cod.removeAllListeners();
+        this.cod.on('open', () => {
+            this.connected = true;
+        });
+        this.cod.on('close', () => {
+            if (this.connected) {
+                this.rm.clear();
+                this.connect();
```

Comment:
Tady bys asi nemel volat connect pokud se to dela uz interne v cod?

Reply:
!